Src/Common/W3C/W3CActivityExtensions.cs

```diff
@@ -0,0 +1,235 @@
+#endif
+    static class W3CActivityExtensions
+    {
+        /// <summary>
+        /// Generate new W3C context.
+        /// </summary>
+        /// <param name="activity">Activity to generate W3C context on.</param>
+        /// <returns>The same Activity for chaining.</returns>
+        [Obsolete("Not ready for public consumption.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static Activity GenerateW3CContext(this Activity activity)
+        {
+            activity.SetVersion(W3CConstants.DefaultVersion);
+            activity.SetSampled(W3CConstants.DefaultSampled);
+            activity.SetSpanId(StringUtilities.GenerateSpanId());
+            activity.SetTraceId(StringUtilities.GenerateTraceId());
+            return activity;
+        }
+
+        /// <summary>
+        /// Checks if current Actuvuty has W3C properties on it.
+        /// </summary>
+        /// <param name="activity">Activity to check.</param>
+        /// <returns>True if Activity has W3C properties, false otherwise.</returns>
+        [Obsolete("Not ready for public consumption.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static bool IsW3CActivity(this Activity activity)
+        {
+            return activity != null && activity.Tags.Any(t => t.Key == W3CConstants.TraceIdTag);
+        }
+
+        /// <summary>
+        /// Updates context on the Activity based on the W3C Context in the parent Activity tree.
+        /// </summary>
+        /// <param name="activity">Activity to update W3C context on.</param>
+        /// <returns>The same Activity for chaining.</returns>
+        [Obsolete("Not ready for public consumption.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static Activity UpdateContextOnActivity(this Activity activity)
+        {
+            if (activity == null || activity.Tags.Any(t => t.Key == W3CConstants.TraceIdTag))
+            {
+                return activity;
+            }
+
+            // no w3c Tags on Activity
+            activity.Parent.UpdateContextOnActivity();
+
+            // at this point, Parent has W3C tags, but current activity does not - update it
+            return activity.UpdateContextFromParent();
+        }
+
+        /// <summary>
+        /// Gets traceparent header value for the Activity or null if there is no W3C context on it.
+        /// </summary>
+        /// <param name="activity">Activity to read W3C context from.</param>
+        /// <returns>traceparent header value.</returns>
+        [Obsolete("Not ready for public consumption.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static string GetTraceParent(this Activity activity)
+        {
+            string version = null, traceId = null, spanId = null, sampled = null;
+            foreach (var tag in activity.Tags)
+            {
+                switch (tag.Key)
+                {
+                    case W3CConstants.TraceIdTag:
```

Comment:
why?

Reply:
This would make it easier for people to read and maintain the code, similar like how we organize the using namespace statements - not a must have but definitely helpful.

Reply:
while this makes sense, I do have them ordered based on the importance and think this is an appropriate order.